sdk/nodejs/resource.ts

```diff
@@ -486,6 +508,33 @@ export interface CustomTimeouts {
     delete?: string;
 }
 
+/**
+ * ResourceTransformation is the callback signature for the `transformations` resource option.  A
+ * transformation is passed the same set of inputs provided to the `Resource` constructor, and can
+ * optionally return back alternate values for the `props` and/or `opts` prior to the resource
+ * actually being created.  The effect will be as though those props and opts were passed in place
+ * of the original call to the `Resource` constructor.  If the transformation returns undefined,
+ * this indicates that the resource will not be transformed.
+ */
+export type ResourceTransformation =
+    (type: string, name: string, props: Inputs, opts: ResourceOptions) => ResourceTransformationResult | undefined;
+
+/**
+ * ResourceTransformationResult is the result that must be returned by a resource transformation
+ * callback.  It includes new values to use for the `props` and `opts` of the `Resource` in place of
+ * the originally provided values.
+ */
+export interface ResourceTransformationResult {
+    /**
+     * The new properties to use in place of the original `props`
+     */
+    props: Inputs;
```

Comment:
Do we envision supporting name transforms eventually? (May help with auto-naming stuff.)

Reply:
Ahh, nevermind, I realize now that you can simply set the physical resource name property. No need to change the logical name.